### ISSUE TITLE
Attempted to extract embedded overlay

### ISSUE THREAD

**Issue by tznind** [fo-dicom/fo-dicom#1385]
Hi,  I am attempting to extract an overlay from a DICOM file.  The file is from the conformance archive of GDCM [gdcmConformanceTests.tar.bz2](https://iweb.dl.sourceforge.net/project/gdcm/gdcmConformanceTests/gdcmConformanceTests/gdcmConformanceTests.tar.bz2).  The image I am having difficulty with is 'XA_GE_JPEG_02_with_Overlays.dcm'.  Any guidance would be much appreciated.

I have written a unit test that shows what I am attempting (see below).  I've left in the image reading code to get the Bmp just to rule out codec issues reading the actual image itself.

**Describe the bug**

The test fails when attempting to access the overlay `Data` field (the last line of the test).  The Exception is:

```
Message: 
FellowOakDicom.Imaging.DicomImagingException : Attempted to extract embedded overlay from compressed pixel data. Decompress pixel data before attempting this operation.

  Stack Trace: 
DicomOverlayData.Load() line 283
DicomOverlayData.get_Data() line 180
Test.Test_XA_GE_JPEG_02_with_Overlays() line 64

  Standard Output: 
 Found Overlay 0 in 'Z:\Repos\fo-dicom\Tests\FO-DICOM.Tests\Bugs\XA_GE_JPEG_02_with_Overlays.dcm'
  Got number of frames 0 in overlay 24576 from 'Z:\Repos\fo-dicom\Tests\FO-DICOM.Tests\Bugs\XA_GE_JPEG_02_with_Overlays.dcm'
  Got old-style overlay 0 in bit position 8 from 'Z:\Repos\fo-dicom\Tests\FO-DICOM.Tests\Bugs\XA_GE_JPEG_02_with_Overlays.dcm'
  Got DicomOverlayData 0 from 'Z:\Repos\fo-dicom\Tests\FO-DICOM.Tests\Bugs\XA_GE_JPEG_02_with_Overlays.dcm'
```

### Unit Test

```csharp
using FellowOakDicom.Imaging;
using FellowOakDicom.Imaging.Render;
using System.IO;
using System.Linq;
using Xunit;
using Xunit.Abstractions;

namespace FellowOakDicom.Tests.Bugs
{
    public class Test
    {
        readonly ITestOutputHelper output;

        public Test(ITestOutputHelper output)
        {
            this.output = output;
        }

        [Fact]
        public void Test_XA_GE_JPEG_02_with_Overlays()
        {
            // file can be found in GDCM conformance tests archive:
            //https://iweb.dl.sourceforge.net/project/gdcm/gdcmConformanceTests/gdcmConformanceTests/gdcmConformanceTests.tar.bz2

            var fi = new FileInfo(@"Z:\Repos\fo-dicom\Tests\FO-DICOM.Tests\Bugs\XA_GE_JPEG_02_with_Overlays.dcm");

            // we can get an image no problem
            var img = new DicomImage(fi.FullName, 0);
            Assert.True(img.Width > 0);
            Assert.True(img.Height > 0);

            new DicomSetupBuilder().RegisterServices(s => s.AddImageManager<ImageSharpImageManager>()).Build();

            var dicomImage = img.RenderImage(0);
            Assert.NotNull(dicomImage);          

            var bmp = dicomImage.AsSharpImage();
            
            Assert.Equal(img.Width, bmp.Width);
            Assert.Equal(img.Height, bmp.Height);
            bmp.Dispose();

            var file = DicomFile.Open(fi.FullName);
            var ds = file.Dataset;

            // Process all of the Overlays
            //   Each overlay can have multiple frames
            var groups =
                ds.Where(x => x.Tag.Group >= 0x6000 && x.Tag.Group <= 0x60FF &&
                x.Tag.Element == 0x0010).Select(x => x.Tag.Group)
                .ToList();

            foreach (var group in groups)
            {
                output.WriteLine($" Found Overlay {group - 0x6000} in '{fi.FullName}'");

                // Check NumberOfFramesInOverlay, if present
                var numframes = ds.GetValueOrDefault<ushort>(new DicomTag(group, 0x0015), 0, 0);
                output.WriteLine($"  Got number of frames {numframes} in overlay {group} from '{fi.FullName}'");

                // Check OverlayBitPosition, normally 0, or bit position for old-style
                var bitpos = ds.GetValue<ushort>(new DicomTag(group, 0x0102), 0);

                if (bitpos > 0)
                    output.WriteLine($"  Got old-style overlay {group - 0x6000} in bit position {bitpos} from '{fi.FullName}'");

                // See https://fo-dicom.github.io/stable/v5/api/FellowOakDicom.Imaging.DicomOverlayData.html
                DicomOverlayData overlay = new DicomOverlayData(ds, group);
                output.WriteLine($"  Got DicomOverlayData {group - 0x6000} from '{fi.FullName}'");

                // This is the line that fails
                Assert.NotNull(overlay.Data);
            }
        }
    }
}
```

### Related code
The stack trace refers to this bit of code:

https://github.com/fo-dicom/fo-dicom/blob/9832ab62de7a79a1a78a19438df69581079f91b9/FO-DICOM.Core/Imaging/DicomOverlayData.cs#L278-L283

**Comment by gofal** [comment]
Well, I assume the issue is exactly what the error message tells you.
I am not able to download the file from the link you posted. Which TransferSyntax does your image have? If the data is compressed then you should decompress it first, by registering a transcodermanager in DicomSetupBuiler and clone the dataset to some uncompressed transfersyntax before accessing the overlays.

**Comment by gofal** [comment]
I now was able to find the files in some other source.
it is just as I wrote in the comment 2 years ago: the overlays are not stored in the current way as a separate dicomtag, but they are stored in the deprecated way, where the unused bits in pixeldata are used. That means, in order to extract the overlay data, the pixeldata has to be known.
the image is compressed in JpegLossless. This means, that the pixeldata cannot be accessed, the data first has to be decompressed (cloned in some uncrompressed transfersyntax), and in order to decode the JpegLossless, a TranscoderManager has to be added to DicomSetupBuilder.
